packages/@aws-cdk/aws-redshift/lib/cluster.ts

```diff
@@ -662,4 +678,69 @@ export class Cluster extends ClusterBase {
       throw new Error('Cannot add a parameter to an imported parameter group.');
     }
   }
+
+  /**
+   * Enables automatic cluster rebooting when changes to the cluster's parameter group require a restart to apply.
+   */
+  public enableRebootForParameterChanges(): void {
+    if (!this.rebootForParameterChangesEnabled) {
+      this.rebootForParameterChangesEnabled = true;
+      const rebootFunction = new lambda.SingletonFunction(this, 'RedshiftClusterRebooterFunction', {
+        uuid: '511e207f-13df-4b8b-b632-c32b30b65ac2',
+        runtime: lambda.Runtime.NODEJS_16_X,
+        code: lambda.Code.fromAsset(path.join(__dirname, 'cluster-parameter-change-reboot-handler')),
+        handler: 'index.handler',
+        timeout: Duration.seconds(900),
+      });
+      rebootFunction.addToRolePolicy(new iam.PolicyStatement({
+        actions: ['redshift:DescribeClusters'],
+        resources: ['*'],
+      }));
+      rebootFunction.addToRolePolicy(new iam.PolicyStatement({
+        actions: ['redshift:RebootCluster'],
+        resources: [
+          Stack.of(this).formatArn({
+            service: 'redshift',
+            resource: 'cluster',
+            resourceName: this.clusterName,
+            arnFormat: ArnFormat.COLON_RESOURCE_NAME,
+          }),
+        ],
+      }));
+      const provider = new cr.Provider(this, 'ResourceProvider', {
+        onEventHandler: rebootFunction,
+      });
+      const customResource = new CustomResource(this, 'RedshiftClusterRebooterCustomResource', {
+        resourceType: 'Custom::RedshiftClusterRebooter',
+        serviceToken: provider.serviceToken,
+        properties: {
+          ClusterId: this.clusterName,
+          ParameterGroupName: Lazy.string({
+            produce: () => {
+              if (!this.parameterGroup) {
+                throw new Error('Cannot enable reboot for parameter changes when there is no associated ClusterParameterGroup.');
```

Comment:
why?

Reply:
Putting implementation details aside I didn't think it would be a good user experience to allow for enabling a feature that does nothing.

Using a `Lazy` gets around an order of operations error where enabling the feature and then later adding a parameter group would trigger this error ([this test case](https://github.com/aws/aws-cdk/pull/22063/files#diff-adc7742a496eece1305edc13496af7b2cfa60b7bf95f14a1852ffa62d438af08R648))